packages/api-client/src/v2/features/operation/components/Header.vue

```diff
@@ -0,0 +1,163 @@
+<script setup lang="ts">
+import type { HttpMethod } from '@scalar/helpers/http/http-methods'
+import type { Environment } from '@scalar/oas-utils/entities/environment'
+import type { ServerObject } from '@scalar/workspace-store/schemas/v3.1/strict/server'
+
+import { OpenApiClientButton } from '@/components'
+import type { ClientLayout } from '@/hooks'
+import type { EnvVariable } from '@/store'
+import type { createStoreEvents } from '@/store/events'
+import { AddressBar, type History } from '@/v2/blocks/scalar-address-bar-block'
+
+const { showSidebar = true, hideClientButton = false } = defineProps<{
+  /** Current request path */
+  path: string
+  /** Current request method */
+  method: HttpMethod
+  /** Client layout */
+  layout: ClientLayout
+
+  /** Controlls sidebar visibility */
+  showSidebar?: boolean
+  /** Hides the client button on the header */
+  hideClientButton?: boolean
+  /** Client integration  */
+  integration?: string | null
+  /** Openapi document url for `modal` mode to open the client app */
+  documentUrl?: string
+  /** Client source */
+  source?: 'gitbook' | 'api-reference'
+
+  /** Currently selected server */
+  server: ServerObject | undefined
+  /** Server list available for operation/document */
+  servers: ServerObject[]
+
+  /** List of request history */
+  history: History[]
+  /**
+   * When the request is sent from the modal, this indicates the progress percentage
+   * of the request being sent.
+   *
+   * The amount remaining to load from 100 -> 0
+   */
+  requestLoadingPercentage?: number
+  /**
+  /** Event bus */
+  events: ReturnType<typeof createStoreEvents>
+
+  /** TODO: to be removed once we fully migrate to the new store */
+  environment: Environment
+  envVariables: EnvVariable[]
+}>()
+
+const emit = defineEmits<{
+  /** Address bar events */
+  (e: 'importCurl', value: string): void
+  (e: 'update:method', payload: { method: HttpMethod }): void
+  (e: 'update:path', payload: { path: string }): void
+  (e: 'execute'): void
+  (e: 'update:selectedServer', payload: { id: string }): void
+  (e: 'update:variable', payload: { key: string; value: string }): void
+  (e: 'addServer'): void
```

Comment:
```suggestion
  (e: 'add:server'): void
```

lets keep this syntax for event names where we touch the store
action:subject

action = add/delete/update